core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -22,43 +22,56 @@
 import org.apache.lucene.search.FuzzyQuery;
 import org.apache.lucene.util.automaton.LevenshteinAutomata;
 import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentBuilderString;
 import org.elasticsearch.common.xcontent.XContentParser;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  * A unit class that encapsulates all in-exact search
  * parsing and conversion from similarities to edit distances
  * etc.
  */
-public final class Fuzziness implements ToXContent {
+public final class Fuzziness implements ToXContent, Writeable<Fuzziness> {
 
     public static final XContentBuilderString X_FIELD_NAME = new XContentBuilderString("fuzziness");
     public static final Fuzziness ZERO = new Fuzziness(0);
     public static final Fuzziness ONE = new Fuzziness(1);
     public static final Fuzziness TWO = new Fuzziness(2);
     public static final Fuzziness AUTO = new Fuzziness("AUTO");
+    
     public static final ParseField FIELD = new ParseField(X_FIELD_NAME.camelCase().getValue());
 
+    /** the prototype constant is intended for deserialization when used with 
+     * {@link org.elasticsearch.common.io.stream.StreamableReader#readFrom(StreamInput)} */
+    public static final Fuzziness PROTOTYPE = AUTO;
+    
     private final Object fuzziness;
 
     private Fuzziness(int fuzziness) {
+        this((Object) fuzziness);
         Preconditions.checkArgument(fuzziness >= 0 && fuzziness <= 2, "Valid edit distances are [0, 1, 2] but was [" + fuzziness + "]");
-        this.fuzziness = fuzziness;
     }
 
     private Fuzziness(float fuzziness) {
+        this((Object) fuzziness);
         Preconditions.checkArgument(fuzziness >= 0.0 && fuzziness < 1.0f, "Valid similarities must be in the interval [0..1] but was [" + fuzziness + "]");
-        this.fuzziness = fuzziness;
```

Comment:
I think I would revert this change, no need to call the other constructor here, so the precondition can happen first, which makes more sense. Same in other constructors